crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,308 @@
+                    message: "Is not stable".to_owned(),
+                    advisory: Advisory::Improvement,
+                }),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that checks if an attribute is in snake case
+pub struct CorrectCaseAdvisor;
+impl Advisor for CorrectCaseAdvisor {
+    fn advise(
+        &self,
+        attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        // Don't provide advice if the attribute is a match
+        if semconv_attribute.is_some() {
+            return None;
+        }
+
+        let is_snake_case = attribute
+            .name
+            .without_boundaries(&Boundary::defaults())
+            .to_case(Case::Snake)
+            == attribute.name;
+        if !is_snake_case {
+            Some(Advice {
+                key: "correct_case".to_owned(),
+                value: Value::Bool(false),
+                message: "Is not in snake case".to_owned(),
+                advisory: Advisory::Violation,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that checks if an attribute has a namespace - a prefix before the first dot
+pub struct NamespaceAdvisor;
+impl Advisor for NamespaceAdvisor {
+    fn advise(
+        &self,
+        attribute: &SampleAttribute,
+        health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        // Don't provide advice if the attribute is a match
+        if semconv_attribute.is_some() {
+            return None;
+        }
+
+        if let Some(last_dot_pos) = attribute.name.rfind('.') {
+            let namespace = attribute.name[..last_dot_pos].to_string();
+
+            // Has a namespace that matches an existing attribute
+            if let Some(found_attr) = health_checker.find_attribute_from_namespace(&namespace) {
+                return Some(Advice {
+                    key: "namespace".to_owned(),
+                    value: Value::String(found_attr.name.clone()),
+                    message: "Namespace matches existing attribute".to_owned(),
+                    advisory: Advisory::Violation,
+                });
+            }
+
+            // Extends an existing namespace
+            if let Some(existing_namespace) = health_checker.find_namespace(&namespace) {
+                return Some(Advice {
+                    key: "namespace".to_owned(),
+                    value: Value::String(existing_namespace),
+                    message: "Extends existing namespace".to_owned(),
+                    advisory: Advisory::Information,
+                });
+            }
+        } else {
+            // Does not have a namespace
+            return Some(Advice {
+                key: "namespace".to_owned(),
```

Comment:
Why isn't this a `no_namespace` key? This would provide a way to filter by advisor output type. I have the same kind of comment for the other advisor results related to the namespace concept.

Reply:
See https://github.com/open-telemetry/weaver/pull/630/commits/7a86e4c5ccc32d00a5bef9265eb629e285f7a4c0